### ISSUE TITLE
Folder selection should be a tree

### ISSUE THREAD

**Issue by jkufner** [PhotoBackup/client-android#50]
When selecting directories to upload the list is not very practical, because there is a lot of stuff on the list and it is hard to find what should be synced. The selection works fine, but the list should preserve directory hierarchy and allow to collapse the subtrees.

**Comment by stephanepechard** [comment]
This is not that easy, as the Android media store used to query data on
pictures does not reflect the file system as you see it when you browse it.
In fact, the term 'folder' is not correct and may induce false
interpretation. The technical term is 'bucket'. I thought it was not
understandable enough, but I may be wrong if users expect to see the file
system directories.

**Comment by jkufner** [comment]
I would definitely expect to see filesystem directories. It is even more confusing to see DCIM directory in file manager and not see it in the bucket list.

**Comment by stephanepechard** [comment]
The DCIM folder is the Camera bucket (at least on my phone), which is quite
understandable.
My main concern over the use of the full file system is the UI which is
cluttered when displaying the classical Android tree. Selecting folders in
such a mess is suboptimal.

**Comment by jkufner** [comment]
Yes, the "DCIM" is understandable, but it is missing in the list when there is no photo directly in the DCIM directory. So if a camera stores photos in DCIM/Camera, then DCIM does not show in bucket list.

The bucket list on my phone has about 100 items.

Selecting folders is completely fine when the folders are collapsible. Then you will see only those few folder you are interested in, usually root directories of both SD cards with DCIM directories selected.
